sympy/simplify/simplify.py

```diff
@@ -1101,6 +1106,53 @@ def bottom_up(rv, F, atoms=False, nonbasic=False):
     return rv
 
 
+def kroneckersimp(expr):
+    """
+    Simplify expressions with KroneckerDelta.
+
+    The only simplification currently attempted is to identify multiplicative cancellation:
+
+    >>> from sympy import KroneckerDelta, kroneckersimp
+    >>> from sympy.abc import i, j
+    >>> kroneckersimp(1 + KroneckerDelta(0, j) * KroneckerDelta(1, j))
+    1
+    """
+    def args_cancel(a1, a2, a3, a4):
+        return Eq(a1, a2) is S.true and Eq(a3, a4) is S.false
+
+    def cancel_kronecker_mul(m):
+        args = m.args
+        deltas = [a for a in args if isinstance(a, KroneckerDelta)]
+
+        for i in range(len(deltas)):
+            args1 = deltas[i].args
+            for j in range(i+1, len(deltas)):
+                args2 = deltas[j].args
+                for i1 in range(2):
+                    for i2 in range(2):
+                        a1 = args1[i1]
+                        a2 = args2[i2]
+                        a3 = args1[(i1 + 1) % 2]
+                        a4 = args2[(i2 + 1) % 2]
+                        if args_cancel(a1, a2, a3, a4):
+                            return Mul(*((0,) + args))
```

Comment:
It may simply be zero

Reply:
We have to be careful about infinities I tihink

Reply:
But since the two terms are rewritten as zero, `args` may have to exclude those two

Reply:
They will be ignored anyway because
```
In [12]: 0*KroneckerDelta(n, m)                                                                          
Out[12]: 0
```
Actually this line could just be `return 0*m`.